src/instances.ts

```diff
@@ -507,7 +507,11 @@ function getOutputFileNames(
     }
     if (
       (configFile.options.declaration || configFile.options.composite) &&
-      (instance.compiler as any).hasTSFileExtension(inputFileName)
```

Comment:
You want to add this part to `getEmitOutput` to return [] early even before getting program rather than here.

Reply:
done, thank you